specs/fault-proof/stage-one/optimism-portal.md

```diff
@@ -0,0 +1,251 @@
+
+### L2 Withdrawal Sender
+
+The **L2 Withdrawal Sender** is the address of the account that triggered a given withdrawal
+transaction on L2. The `OptimismPortal` is expected to expose a variable that includes this value
+when [finalizing](#finalized-withdrawal) a withdrawal.
+
+## Assumptions
+
+### aOP-001: Dispute Game contracts properly report important properties
+
+We assume that the `FaultDisputeGame` and `PermissionedDisputeGame` contracts properly and
+faithfully report the following properties:
+
+- Game type
+- L2 block number
+- Root claim value
+- Game extra data
+- Creation timestamp
+- Resolution timestamp
+- Resolution result
+- Whether the game was the respected game type at creation
+
+We also specifically assume that the game creation timestamp and the resolution timestamp are not
+set to values in the future.
+
+#### Mitigations
+
+- Existing audit on the `FaultDisputeGame` contract
+- Integration testing
+
+### aOP-002: DisputeGameFactory properly reports its created games
+
+We assume that the `DisputeGameFactory` contract properly and faithfully reports the games it has
+created.
+
+#### Mitigations
+
+- Existing audit on the `DisputeGameFactory` contract
+- Integration testing
+
+### aOP-003: Incorrectly resolving games will be invalidated within the airgap delay period
+
+We assume that any games that are resolved incorrectly will be invalidated within the airgap delay
+period.
+
+#### Mitigations
+
+- Stakeholder incentives / processes
+- Incident response plan
+- Monitoring
+
+## Invariants
+
+### iOP-001: Invalid Withdrawals can never be finalized
+
+We require that [Invalid Withdrawals](#invalid-withdrawal) can never be
+[finalized](#finalized-withdrawal) for any reason.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is broken, any number of arbitrarily bad outcomes could happen. Most obviously,
+we would expect all bridge systems relying on the `OptimismPortal` to be immediately compromised.
+
+### iOP-002: Valid Withdrawals can always be finalized in bounded time
+
+We require that [Valid Withdrawals](#valid-withdrawal) can always be
+[finalized](#finalized-withdrawal) within some reasonable, bounded amount of time.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is broken, we would expect that users are unable to withdraw bridged assets. We
+see this as a critical system risk.
+
+## Function Specification
+
+### setRespectedGameType
+
+Allows the Guardian to change the [Respected Game Type](#respected-game-type) and to update the
+[Retirement Timestamp](#retirement-timestamp).
+
+- MUST revert if the sender is not the Guardian.
+- If the game type is `type(uint32).max`, MUST update the Retirement Timestamp to the current
+  block timestamp and MUST NOT update the Respected Game Type.
+- If the game type is not `type(uint32).max`, MUST set the Respected Game Type to the provided game
+  type and MUST NOT update the Retirement Timestamp.
```

Comment:
This is a very weird API. Not a blocker, but I'm very surprised we didn't just introduce a separate method for updating the retirement timestamp rather than using a magic value to shoehorn it into the same method.

Reply:
Ahh so the actual reason here was that changing the API or adding a new function meant that we would have to redeploy the `DeputyGuardianModule` which would've been a PITA (and would've required new pre-signed pause transactions).

We *will* be removing this weird API as part of the very next upgrade. See [here](https://github.com/ethereum-optimism/optimism/pull/13824) for a draft implementation of the next upgrade. Specs coming soon, I promise it will fix this!